### PR TITLE
`readConcern`

### LINKED ISSUE #14511: be able to set readConcern on mongoose.Schema
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Be able to define a readConcern similar to writeConcern at a mongoose.Schema level

### Motivation

Simplify the use of readConcern per schema

### Example

```tsx
const Schema = new mongoose.Schema(
  {    
  },
  {
    timestamps: true,
    collection: 'LedgerEntry',
    writeConcern,
    readConcern,
  },
);
```

### PR DESCRIPTION
**Pull request by IslandRhythms**
closes #14511

### CODE REVIEW

File: lib/helpers/schema/applyReadConcern.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  // Don't apply default write concern to operations in transactions,
+  // because setting write concern on an operation in a transaction is an error
+  // See: https://www.mongodb.com/docs/manual/reference/write-concern/
+  if (options && options.session && options.session.transaction) {
+    return;
+  }
+  const readConcern = get(schema, 'options.readConcern', {});
+  if (Object.keys(readConcern).length != 0) {
+    options.readConcern = {};
+    if (!('majoriy' in options) && readConcern.level != null) {
```
Comment (vkarpov15) [inline lib/helpers/schema/applyReadConcern.js:18]: "majoriy"?

File: lib/helpers/schema/applyReadConcern.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      options.readConcern.level = readConcern.level;
+    }
+  }
+  else {
+    if (!('majoriy' in options) && readConcern.level != null) {
```
Comment (vkarpov15) [inline lib/helpers/schema/applyReadConcern.js:23]: "majoriy"?

File: lib/helpers/schema/applyReadConcern.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    return;
+  }
+  const readConcern = get(schema, 'options.readConcern', {});
+  if (Object.keys(readConcern).length != 0) {
+    options.readConcern = {};
+    if (!('majoriy' in options) && readConcern.level != null) {
+      options.readConcern.level = readConcern.level;
+    }
+  }
+  else {
```
Comment (vkarpov15) [inline lib/helpers/schema/applyReadConcern.js:22]: This `else {}` statement does the exact same thing as the `if {}` branch above, please get rid of it

File: lib/query.js [hunk 4]
```diff
@@ -1311,6 +1312,19 @@ Query.prototype.writeConcern = function writeConcern(val) {
   return this;
 };
 
+/*
+Query.prototype.readConcern = function readConcern(val) {
```
Comment (vkarpov15) [inline lib/query.js:1316]: Get rid of this commented out code

File: test/helpers/applyReadConcern.test.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const assert = require('assert');
+const applyReadConcern = require('../../lib/helpers/schema/applyReadConcern');
+const start = require('../common');
+const mongoose = start.mongoose;
+
+describe('applyReadConcern', function() {
```
Comment (vkarpov15) [inline test/helpers/applyReadConcern.test.js:8]: Also add a test case in `query.test.js` that calls `_optionsForExec()` and makes sure that 1) schema-level read concern is applied if user doesn't pass in `readConcern` option, 2) calling `Model.find().setOptions({ readConcern: { level: 'majority' } })` overrides schema-level read concern

**Review by vkarpov15 — CHANGES_REQUESTED** [review]

File: lib/helpers/schema/applyReadConcern.js (hunk 3 above)
Comment (IslandRhythms) [reply lib/helpers/schema/applyReadConcern.js:22]: should I do the same for `applyWriteConcern` then?

**Comment by vkarpov15** [comment]
I'm just gonna close this PR and start from scratch, the work here is headed in the wrong direction

**Pull request closed without merging**